### ISSUE TITLE
Issue on Arch Linux ARM

### ISSUE THREAD

**Issue by pezz** [grazzolini/mkinitcpio-tinyssh#3]
Hi again mate,

I've been setting this up on all of my boxes, and had no issues until I set it up under Arch Linux ARM (Raspberry Pi 2, Model B - armv7h).

I've just debugged it for a while now, and when run from an initrd on Arch ARM it looks for authorized_keys in //.ssh

Not sure if this bug is yours to deal with, a tinyssh bug, or something the Arch ARM guys aren't setting properly in early userspace.

Basically, /root/.ssh is not looked at, I believe, because the root home dir is set to / and /.ssh is the expected directory. /root/.ssh is correctly added to the initrd but is not used.

Hope this makes sense.

Cheers.

**Comment by pezz** [comment]
FYI - workaround to make it work on Arch ARM:

mkdir /.ssh
chmod 700 /.ssh
cp your .pub key to /.ssh/authorized_keys

Add to FILES in /etc/mkinitcpio.conf:
FILES="/.ssh/authorized_keys"

mkinitcpio -k 4.1.10-2-ARCH -g /boot/initrd

Or however you build your initrd...

**Comment by grazzolini** [comment]
It seems your initrd /etc/passwd isn't being generated. Are you using the encryptssh hook from the https://github.com/grazzolini/mkinitcpio-utils package? It is this hook that installs the cryptsetup_shell needed for remote unlocking and also creates the passwd file.

**Comment by pezz** [comment]
Yes, I am using all of your hooks. I extracted the passwd file from the created initrd and it contains:

`root:x:0:0:root:/:/bin/cryptsetup_shell`

So I guess root is getting a home dir of /

**Comment by grazzolini** [comment]
Check the first few lines of the encryptssh hook in mkinitcpio-utils:

https://github.com/grazzolini/mkinitcpio-utils/blob/master/initcpio/install/encryptssh

It is clearly creating the /etc/passwd file with the root home under /root. Check your /usr/lib/initcpio/install/encryptssh file and see if it's not the old version, the one bundled with dropbear_initrd_encrypt. If it is, then you need to make sure you have all three packages installed: mkinitcpio-netconf, mkinitcpio-dropbear and mkinitcpio-utils. And no dropbear_initrd_encrypt.

**Comment by pezz** [comment]
You're dead right, /usr/lib/initcpio/install/encryptssh did have / specified.

I've re-installed mkinitcpio-utils and it now correctly defines /root as the home dir.

Very odd that my system didn't have the updated version.

Anyway, thanks for the help, nothing to fix here. Cheers.
